src/Sitegen.Tests/ProgramTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using FluentAssertions;
+using Sitegen.Models;
+using Sitegen.Models.Config;
+using Xunit;
+
+namespace Sitegen.Tests
+{
+    public static class ReadConfig
+    {
+        public class WithEmptyConfig
+        {
+            [Fact]
+            void throws_expected_exception()
+            {
+                Action action = () => Program.ReadConfig("fixtures/empty_config.yaml");
+
+                action.Should()
+                    .Throw<ConfigurationException>()
+                    .WithMessage("config.yaml does not contain any settings");
+            }
+        }
```

Comment:
This illustrates my point in https://github.com/perlun/sitegen/pull/54/files#r670768803. I want to be able to group test methods like this ("With empty config", "With non existent config") and have some shared state between test methods for these grouped test methods. It's really useful once you've gotten used to it.